back/app/services/slug_service.rb

```diff
@@ -1,17 +1,18 @@
 class SlugService
 
-  def generate_slug record, title
-    return nil if !title
-    slug = slugify(title)
+  def generate_slug record, string
+    return nil if !string
+
+    slug = slugify(string)
     indexedSlug = nil
     i=0
     while record.class.find_by(slug: indexedSlug || slug)
       i +=1
       indexedSlug = [slug, '-', i].join
     end
-    indexedSlug || slug
-  end
 
+    return indexedSlug || slug
+  end
```

Comment:
The next method (below this comment), `generate_slugs` is currently only called by `UserSlugService#generate_slugs`, as the only time we generate more than one slug (i.e. in bulk) is when we create user slugs when creating invites in bulk from an imported XLSX.

I decided to leave this method here as it might be useful in the future if we decide to bulk create slugs for any object other than Users, and would save the dev the effort of working out how to refactor out the common process.

I have no strong feelings about this, however, and would be happy to move this into `UserSlugService#generate_slugs` if that is preferred.